### PR TITLE
TYP: use DTypeLike alias from numpy.typing

### PR DESCRIPTION
**Pull request by simonjayhawkins**

### CODE REVIEW

File: pandas/_typing.py [hunk 1]
```diff
@@ -122,10 +129,7 @@
 Axes = Collection[Any]
 
 # dtypes
-NpDtype = Union[str, np.dtype]
-Dtype = Union[
-    "ExtensionDtype", NpDtype, type_t[Union[str, float, int, complex, bool, object]]
-]
+Dtype = Union["ExtensionDtype", NpDtype]
```
Comment (jbrockmendel) [inline pandas/_typing.py:132]: do we need to worry about strings like "categorical" that can be interpreted as pandas dtypes but not np.dtype?

Comment (Dr-Irv) [reply pandas/_typing.py:132]: I think we are only using `NpDtype` where we are expecting Numpy dtypes

**Comment by Dr-Irv** [comment]
@simonjayhawkins I hope you an approve and merge in #40421 before this one gets merged, because many of the changes there are with respect to using `NpDtype` in the right places with respect to the `to_numpy()` calls in `ExtensionArray` classes and subclasses.

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins I hope you an approve and merge in #40421 before this one gets merged, because many of the changes there are with respect to using `NpDtype` in the right places with respect to the `to_numpy()` calls in `ExtensionArray` classes and subclasses.

I recall suggesting doing this as a precursor. some changes in #40421 would become unnecessary as the mypy errors are false postives.

**Comment by simonjayhawkins** [comment]
closing to clear queue. will hopefully circle back to typing next week.

**Comment by simonjayhawkins** [comment]
@Dr-Irv @jbrockmendel I probably won't have time to revisit this for a while (probably after 1.3.1 to allow time to follow regressions from 1.3) so feel free to take this on.

**Comment by simonjayhawkins** [comment]
I could be wrong, but I think this is necessary precursor to the EA typing work.

**Pull request closed without merging**